pkg/tests/end_to_end_tests/tag_op_test.go

```diff
@@ -0,0 +1,460 @@
+	{
+		inputType:   "float8",
+		inputValue:  `'-infinity'::float8`,
+		outputType:  "string",
+		outputValue: `"-Infinity"`,
+	},
+	{
+		inputType:   "float8",
+		inputValue:  `'infinity'::float8`,
+		outputType:  "string",
+		outputValue: `"Infinity"`,
+	},
+	{
+		inputType:   "float8",
+		inputValue:  `'NaN'::float8`,
+		outputType:  "string",
+		outputValue: `"NaN"`,
+	},
+	{
+		inputType:   "numeric",
+		inputValue:  `1.0::numeric`,
+		outputType:  "number",
+		outputValue: `1.0`,
+	},
+	{
+		inputType:   "numeric",
+		inputValue:  `1.1::numeric`,
+		outputType:  "number",
+		outputValue: `1.1`,
+	},
+	{
+		inputType:   "numeric",
+		inputValue:  `1.5::numeric`,
+		outputType:  "number",
+		outputValue: `1.5`,
+	},
+	{
+		inputType:   "numeric",
+		inputValue:  `1.99::numeric`,
+		outputType:  "number",
+		outputValue: `1.99`,
+	},
+	{
+		inputType:   "date",
+		inputValue:  `'2021-11-18'::date`,
+		outputType:  "string",
+		outputValue: `"2021-11-18"`,
+	},
+	{
+		inputType:   "date",
+		inputValue:  `'2021-01-01'::date`,
+		outputType:  "string",
+		outputValue: `"2021-01-01"`,
+	},
+	{
+		inputType:   "date",
+		inputValue:  `'2021-12-31'::date`,
+		outputType:  "string",
+		outputValue: `"2021-12-31"`,
+	},
+	{
+		inputType:   "time",
+		inputValue:  `'01:02'::time`,
+		outputType:  "string",
+		outputValue: `"01:02:00"`,
+	},
+	{
+		inputType:   "time",
+		inputValue:  `'12:34:56'::time`,
+		outputType:  "string",
+		outputValue: `"12:34:56"`,
+	},
+	{
+		inputType:   "time",
+		inputValue:  `'23:59:59'::time`,
+		outputType:  "string",
+		outputValue: `"23:59:59"`,
+	},
+	{
+		inputType:   "timestamptz",
+		inputValue:  `'2021-11-18 01:02 UTC'::timestamptz`,
+		outputType:  "string",
+		outputValue: `"2021-11-18T01:02:00+00:00"`,
+	},
+	{
+		inputType:   "timestamptz",
+		inputValue:  `'2021-01-01 12:34:56 UTC'::timestamptz`,
+		outputType:  "string",
+		outputValue: `"2021-01-01T12:34:56+00:00"`,
+	},
+	{
+		inputType:   "timestamptz",
+		inputValue:  `'2021-12-31 23:59:59 UTC'::timestamptz`,
+		outputType:  "string",
+		outputValue: `"2021-12-31T23:59:59+00:00"`,
+	},
+	{
+		inputType:   "interval",
+		inputValue:  `interval '1 second'`,
+		outputType:  "string",
+		outputValue: `"00:00:01"`,
+	},
+	{
+		inputType:   "interval",
+		inputValue:  `interval '1 minute'`,
+		outputType:  "string",
+		outputValue: `"00:01:00"`,
+	},
+	{
+		inputType:   "interval",
+		inputValue:  `interval '1 hour'`,
+		outputType:  "string",
+		outputValue: `"01:00:00"`,
+	},
+	{
+		inputType:   "text",
+		inputValue:  `'hello world'::text`,
+		outputType:  "string",
+		outputValue: `"hello world"`,
+	},
+}
+
+func TestTagOpExpressions(t *testing.T) {
+	var ctx = context.Background()
+	databaseName := fmt.Sprintf("%s_tag_op", *testDatabase)
+	withDB(t, databaseName, func(db *pgxpool.Pool, tb testing.TB) {
+		testTagOperators(t, db, ctx)
+		testTagOpRegexpMatches(t, db, ctx)
+		testTagOpRegexpNotMatches(t, db, ctx)
+		testTagOpJsonPathExists(t, db, ctx)
```

Comment:
instead of passing down `t`, i think it will be better with passing `tb`, since then the error/panic trace will have a more accurate trace.

Reply:
Unfortunately, there is no `Run` method on the `TB` -- only on `T`. I need the `Run` method so I can execute each test as a subtest.